synapse/http/request_metrics.py

```diff
@@ -43,6 +43,28 @@
     "synapse_http_server_response_time_seconds",
```

Comment:
> Is there a particular insight we're hoping to gain by raising the 10 second cap?

I'm trying to optimize the slow `/messages` requests, https://github.com/matrix-org/synapse/issues/13356, specifically those that take more than 10s.

In order to track progress there, I'd like the metrics to capture them.